src/core/diags/factory_diags.cpp

```diff
@@ -942,6 +945,13 @@ void Diags::TransmitDone(Error aError)
     }
     else if (mTxPackets > 1)
     {
+#ifdef OPENTHREAD_CONFIG_PLATFORM_USEC_TIMER_ENABLE
+        // A short delay to give RCP time to process the previous transaction.
+        constexpr uint32_t kInterTransactionDelayUsec = 50;
+        uint32_t start = otPlatAlarmMicroGetNow();
+
+        while ((otPlatAlarmMicroGetNow() - start) < kInterTransactionDelayUsec){};
```

Comment:
Thanks, @clementLH14 for the PR but there are some concerns with this PR as is.

- We cannot and should not use busy-loop waits in OpenThread; that is not acceptable.
- You mention the RCP arhcietcture. The factory diag module is general-purpose and can be used with any architecture, not necessarily the RCP model.
-  It is not clear why the radio or RCP cannot accept a new command after it signals "TxDone". This seems like a radio or RCP serial interface issue. 
    - My suggestion is that the root cause of this should be investigated and addressed. Adding a short delay in factory diag does not seem to be the right thing to do (seems like a hack to mask the issue).